internal/pkg/utils/deepcopy/deepcopy.go

```diff
@@ -116,36 +121,40 @@ func translateRecursive(clone, original reflect.Value, callback TranslateFunc, s
 		// points to, so we have to call Elem() to unwrap it
 		cloneValue := reflect.New(originalValue.Type()).Elem()
 		steps := steps.Add(kind.String(), ``)
-		translateRecursive(cloneValue, originalValue, callback, steps)
+		translateRecursive(cloneValue, originalValue, callback, steps, visited)
 		clone.Set(cloneValue)
 
 	// If it is a struct we translate each field
 	case kind == reflect.Struct:
-		t := original.Type()
+		t := originalType
 		for i := 0; i < original.NumField(); i += 1 {
 			steps := steps.Add(t.String(), t.Field(i).Name)
-			translateRecursive(clone.Field(i), original.Field(i), callback, steps)
+			cloneField := clone.Field(i)
+			if !cloneField.CanSet() {
+				panic(fmt.Errorf("deepcopy found unexported field:\nsteps: %s\nvalue:%#v", steps.String(), original.Interface()))
```

Comment:
pridana detekcia unexported fields, .... ono to spadne aj tak, ale takto aspon vieme kde je problem.